internal/service/anycast/api_anycast_host_resource_test.go

```diff
@@ -0,0 +1,385 @@
+					testAccCheckOnPremAnycastHostExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "config_ospf.area_type", "NSSA"),
+					resource.TestCheckResourceAttr(resourceName, "config_ospf.area", "10.10.0.2"),
+					resource.TestCheckResourceAttr(resourceName, "config_ospf.authentication_type", "MD5"),
+					resource.TestCheckResourceAttr(resourceName, "config_ospf.interface", "ens160"),
+					resource.TestCheckResourceAttr(resourceName, "config_ospf.authentication_key", "YXV0aGV"),
+					resource.TestCheckResourceAttr(resourceName, "config_ospf.hello_interval", "20"),
+					resource.TestCheckResourceAttr(resourceName, "config_ospf.dead_interval", "50"),
+					resource.TestCheckResourceAttr(resourceName, "config_ospf.retransmit_interval", "10"),
+					resource.TestCheckResourceAttr(resourceName, "config_ospf.transmit_delay", "2"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func testAccCheckOnPremAnycastHostExists(ctx context.Context, resourceName string, v *anycast.OnpremHost) resource.TestCheckFunc {
+	// Verify the resource exists in the cloud
+	return func(state *terraform.State) error {
+		rs, ok := state.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+		id, err := strconv.ParseInt(rs.Primary.ID, 10, 64)
+		if err != nil {
+			return fmt.Errorf("error parsing ID: %v", err)
+		}
+		apiRes, _, err := acctest.BloxOneClient.AnycastAPI.
+			OnPremAnycastManagerAPI.
+			GetOnpremHost(ctx, id).
+			Execute()
+		if err != nil {
+			return err
+		}
+		if !apiRes.HasResults() {
+			return fmt.Errorf("expected result to be returned: %s", resourceName)
+		}
+		*v = apiRes.GetResults()
+		return nil
+	}
+}
+
+func testAccCheckOnPremAnycastHostDestroy(ctx context.Context, v *anycast.OnpremHost) resource.TestCheckFunc {
+	// Verify the resource was destroyed
+	return func(state *terraform.State) error {
+		_, httpRes, err := acctest.BloxOneClient.AnycastAPI.
+			OnPremAnycastManagerAPI.
+			GetOnpremHost(ctx, *v.Id).
+			Execute()
+		if err != nil {
+			if httpRes != nil && httpRes.StatusCode == http.StatusNotFound {
+				// resource was deleted
+				return nil
+			}
+			return err
+		}
+		return errors.New("expected to be deleted")
+	}
+}
+
+func testAccCheckOnPremAnycastHostDisappears(ctx context.Context, v *anycast.OnpremHost) resource.TestCheckFunc {
+	// Delete the resource externally to verify disappears test
+	return func(state *terraform.State) error {
+		_, _, err := acctest.BloxOneClient.AnycastAPI.
+			OnPremAnycastManagerAPI.
+			DeleteOnpremHost(ctx, *v.Id). //testAccCheckOnPremAnycastHostDisappears
+			Execute()
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+}
+
+func testAccBaseWithAnycastConfig(name, anycastIpAddress string) string {
+	return fmt.Sprintf(`
+data "bloxone_infra_hosts" "anycast_hosts" {
+  filters = {
+    display_name = "anycast_real"
+  }
+}
```

Comment:
can you try using the resources to create a host and service in this test itself, instead of relying on an actual onprem?

Reply:
done